### PR TITLE
DE TN Fix for Issue #228

### PR DESCRIPTION
**Pull request by zoobereq**
# What does this PR do ?

The current PR continues to address Issue #228 for the German TN graph.  To that end, it does the following:
- Expands the common-domain list to include the failing cases enumerated in the bug report. By doing so, it also signals how the list may be expanded to incorporate other commonly-used domains.
- Expands the list of top-level domains based on Google's domain registry.
- Fixes the issue where the sentence-end period following a URL would be separated from it by a space.

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Have you signed your commits? Use ``git commit -s`` to sign.
- [x] Do all unittests finish successfully before sending PR?
   1) ``pytest`` or (if your machine does not have GPU) ``pytest --cpu`` from the root folder (given you marked your test cases accordingly `@pytest.mark.run_only_on('CPU')`).
   2) Sparrowhawk tests ``bash tools/text_processing_deployment/export_grammars.sh --MODE=test ...``
- [x] If you are adding a new feature: Have you added test cases for both `pytest` and Sparrowhawk [here](tests/nemo_text_processing).
- [x] Have you added ``__init__.py`` for every folder and subfolder, including `data` folder which has .TSV files?
- [x] Have you followed codeQL results and removed unused variables and imports (report is at the bottom of the PR in github review box) ?
- [x] Have you added the correct license header `Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.` to all newly added Python files?
- [x] If you copied [nemo_text_processing/text_normalization/en/graph_utils.py](nemo_text_processing/text_normalization/en/graph_utils.py) your header's second line should be `Copyright 2015 and onwards Google, Inc.`. See an example [here](https://github.com/NVIDIA/NeMo-text-processing/blob/main/nemo_text_processing/text_normalization/en/graph_utils.py#L2).
- [x] Remove import guards (`try import: ... except: ...`) if not already done.
- [x] If you added a new language or a new feature please update the [NeMo documentation](https://github.com/NVIDIA/NeMo/blob/main/docs/source/nlp/text_normalization/wfst/wfst_text_normalization.rst) (lives in different repo).
- [x] Have you added your language support to [tools/text_processing_deployment/pynini_export.py](tools/text_processing_deployment/pynini_export.py).
  

**PR Type**:
- [ ] New Feature
- [x] Bugfix
- [ ] Documentation
- [x] Test

If you haven't finished some of the above items you can still open "Draft" PR.

### CODE REVIEW

File: tests/nemo_text_processing/de/data_text_normalization/test_cases_electronic.txt [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
 a eins b zwei at a b c punkt com~a1b2@abc.com
 a b drei bindestrich s d d bindestrich drei at g mail punkt com~ab3-sdd-3@gmail.com
 h t t p s doppelpunkt slash slash w w w punkt a b c punkt com~https://www.abc.com
 w w w punkt a b c punkt com~www.abc.com
-b r e t t s p i e l v e r s a n d punkt de .~brettspielversand.de.
-w w w punkt e n v e e d y a punkt net .~www.enveedya.net.
-w w w punkt a m a z o n punkt com punkt de .~www.amazon.com.de.
+brettspielversand punkt de.~brettspielversand.de.
+w w w punkt e n v e e d y a punkt net.~www.enveedya.net.
+w w w punkt amazon punkt com punkt de.~www.amazon.com.de.
```
Comment (tbartley94) [inline tests/nemo_text_processing/de/data_text_normalization/test_cases_electronic.txt:11]: extra period intended?

**Review by tbartley94 — CHANGES_REQUESTED** [review]
Will approve once we finalize which class we're adding changes to.

File: tests/nemo_text_processing/de/data_text_normalization/test_cases_electronic.txt (hunk 1 above)
Comment (zoobereq) [reply tests/nemo_text_processing/de/data_text_normalization/test_cases_electronic.txt:11]: I'm not sure I'm following. The periods agree on both sides of the tilde. Unless I'm missing something...

Comment (tbartley94) [reply tests/nemo_text_processing/de/data_text_normalization/test_cases_electronic.txt:11]: I'm just san checking you intended to include the full stop in these examples. It make the electronic test cases become a partial punctuation test case.

Comment (tbartley94) [reply tests/nemo_text_processing/de/data_text_normalization/test_cases_electronic.txt:11]: Nvm i'm seeing it in your draft notes.

**Review by tbartley94 — APPROVED** [review]

**Pull request merged**